editor/src/core/workers/parser-printer/parser-printer-comments.ts

```diff
@@ -2,63 +2,131 @@ import * as TS from 'typescript'
+        hasTrailingNewLine,
+      )
+    default:
+      const _exhaustiveCheck: never = commentKind
+      throw new Error(`Unhandled comment kind ${commentKind}`)
+  }
+}
+
+function getLeadingComments(sourceText: string, node: TS.Node): Array<Comment> {
+  let result: Array<Comment> = []
+  const parseAndPushComment = (
+    pos: number,
+    end: number,
+    commentKind: TS.CommentKind,
+    hasTrailingNewLine: boolean,
+  ) => {
+    result.push(parseComment(sourceText, pos, end, commentKind, hasTrailingNewLine))
   }
+
+  TS.forEachLeadingCommentRange(sourceText, node.getFullStart(), parseAndPushComment)
+  return result
+}
+
+function getTrailingComments(sourceText: string, node: TS.Node): Array<Comment> {
+  let result: Array<Comment> = []
+  const parseAndPushComment = (
+    pos: number,
+    end: number,
+    commentKind: TS.CommentKind,
+    hasTrailingNewLine: boolean,
+  ) => {
+    result.push(parseComment(sourceText, pos, end, commentKind, hasTrailingNewLine))
+  }
+
+  TS.forEachTrailingCommentRange(sourceText, node.getEnd(), parseAndPushComment)
+  return result
+}
+
+export function getComments(sourceText: string, node: TS.Node): ParsedComments {
+  const leadingComments = getLeadingComments(sourceText, node)
+  const trailingComments = getTrailingComments(sourceText, node)
+
+  return parsedComments(leadingComments, trailingComments)
 }
 
-export function addCommentsToNode(node: TS.Node, leadingComments: Array<Comment>): void {
-  let leadingTSComments: Array<TS.SynthesizedComment> = []
-  fastForEach(leadingComments, (leadingComment) => {
+function createTSComments(comments: Array<Comment>): Array<TS.SynthesizedComment> {
+  let result: Array<TS.SynthesizedComment> = []
+  fastForEach(comments, (comment) => {
     let commentKind: TS.CommentKind
-    switch (leadingComment.type) {
+    switch (comment.type) {
       case 'MULTI_LINE_COMMENT':
         commentKind = TS.SyntaxKind.MultiLineCommentTrivia
         break
       case 'SINGLE_LINE_COMMENT':
         commentKind = TS.SyntaxKind.SingleLineCommentTrivia
         break
       default:
-        const _exhaustiveCheck: never = leadingComment
-        throw new Error(`Unhandled comment type ${leadingComment}`)
+        const _exhaustiveCheck: never = comment
+        throw new Error(`Unhandled comment type ${comment}`)
     }
     const tsComment: TS.SynthesizedComment = {
       kind: commentKind,
-      hasTrailingNewLine: leadingComment.trailingNewLine,
-      text: leadingComment.comment,
+      hasTrailingNewLine: comment.trailingNewLine,
+      text: comment.comment,
       pos: -1,
       end: -1,
     }
-    leadingTSComments.push(tsComment)
+    result.push(tsComment)
   })
+  return result
+}
+
+export function addCommentsToNode(node: TS.Node, comments: ParsedComments): void {
+  const leadingTSComments = createTSComments(comments.leadingComments)
+  const trailingTSComments = createTSComments(comments.trailingComments)
+
   if (leadingTSComments.length > 0) {
```

Comment:
Tiny thing, but probably better to check `comments.leadingComments` here to avoid spawning mountains of empty arrays that are immediately thrown away.